convert2rhel/phase.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+class ConversionPhase:
+    """
+    Conversion phase to hold name and logging-friendly name.
+    """
+
+    def __init__(self, name, log_name=None):  # type: (str, str|None) -> None
+        self.name = name
+        self.log_name = log_name
+
+    def __str__(self):
+        return self.log_name if self.log_name else self.name
+
+    def __repr__(self):
+        return self.name
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, value):
+        return self.__hash__() == value.__hash__()
+
+
+class ConversionPhases:
+    """During conversion we will be in different states depending on
+    where we are in the execution. This class establishes the different phases
+    that we have as well as what the current phase is set to.
+    """
+
+    POST_CLI = ConversionPhase(name="POST_CLI")
+    PREPARE = ConversionPhase(name="PREPARE", log_name="Prepare")
```

Comment:
I just wonder, when we are changing the logging and the final is renamed to convert (which makes sense to me) - what about changing the `Prepare` to `Analysis` (or something like that)? I think it would be nice to have it matching the subcommands (analyze and convert).

Reply:
Agreed and I had that before, but I wanted minimal changes in this PR to the actual logs so wanted it to remain the same. We can change it in a follow-up PR but requires a bit more discussion IMO

Reply:
Okay. Maybe if we want minimal changes, then I would change the name of the post_conversion to the `Final` to be consistent